cou/apps/auxiliary.py

```diff
@@ -33,6 +33,37 @@
 class OpenStackAuxiliaryApplication(OpenStackApplication):
     """Application for charms that can have multiple OpenStack releases for a workload."""
 
+    @property
+    def is_os_channel_based(self) -> bool:
+        """Check if application is OpenStack channel based.
+
+        For auxiliary charms, always return false because they are
+        not OpenStack channel based.
+        :return: False.
+        :rtype: bool
+        """
+        return False
+
+    def is_valid_track(self, charm_channel: str) -> bool:
+        """Check if the channel track is valid.
+
+        Auxiliary charms don't follow the OpenStack track convention
+        and are validated based on the openstack_to_track_mapping.csv table.
+        :param charm_channel: Charm channel. E.g: 3.8/stable
+        :type charm_channel: str
+        :return: True if valid, False otherwise.
+        :rtype: bool
+        """
+        if self.is_from_charm_store:
+            logger.debug(
+                "'%s' has been installed from the charm store",
+                self.name,
+            )
+            return True
+
+        track = self._get_track_from_channel(charm_channel)
+        return bool(TRACK_TO_OPENSTACK_MAPPING.get((self.charm, self.series, track)))
```

Comment:
not-blocking: I think that this is not good practice and it could cause an issue. Maybe not in this case, but in general yes.  I think that `(self.charm, self.series, track) in TRACK_TO_OPENSTACK_MAPPING` make much more sense.